### DISCUSSION TITLE
(feature discussion) re-map request body values

### DISCUSSION THREAD

**Discussion by peterbsmyth** [Kong/kong#7250] [Ideas and feature requests]
### Summary

Given a request body, A:  

``` json
{
  "foo": "bar"
}
```

Re-map request to, B:

``` json
{
  "foo": "baz"
}
```

@Tieske @thibaultcha I think this would work well as a part of request-transformer as `map`

**Comment by Tieske** [comment]
Sorry, not getting the intentions here?

**Comment by peterbsmyth** [comment]
Here's a real world scenario:

Kong sits in front of an upstream API whose POST body requires the following: 

``` json
{
    "LOCALE": "en_US"
}
```

`en_US` is not highly user friendly. So it would be nice to remap a more user-friendly request body:

``` json
{
    "LOCALE": "US"
}
```

now the end user of the API can have a nice request body, and the upstream API can receive the body it expects.

**Comment by Tieske** [comment]
ah, but then the requirements should be extended to make it dynamic. See http://nordicapis.com/how-to-control-user-identity-within-microservices/

Case; in the internal network, the token used between microservices is a jwt. This token contains all sorts of user data. Now when a request/response leaves the network, this token should be replaced by an opaque token (something also unique, but not carrying any data). Upon a next request, the opaque token should be replaced again with the one carrying the data.

This way the data never leaves the internal network.

Not sure though whether this is a functionality that has much use on its own (and hence can live in the request-transformer), or whether it is tightly coupled to oauth2, and hence should be part of that plugin.

@thefosk any thoughts?

**Comment by peterbsmyth** [comment]
@Tieske I had never thought of it (for my use-case) as involving Auth at all. I do see your point, however I believe that there is enough functional benefit to adding this to request-transformer (perhaps as "translate") to consider it on those grounds alone.

**Comment by shashiranjan84** [comment]
@peterbsmith2 @Tieske I am still confused, isn't this already handled by the plugin `request-tranformer`, basically by applying the `config.replace` on request json body?

**Comment by Tieske** [comment]
the intent (as I read it) would be to only modify the value if the existing value matches something.

So the example above not a transformation based on key `LOCALE`, but a map on key `LOCALE` where only the value `en_US` would be changed to `US`, and all other values would be left alone.

@peterbsmith2 correct?

**Comment by peterbsmyth** [comment]
@Tieske correct
